src/main/java/randoop/generation/ForwardGenerator.java

```diff
@@ -270,6 +281,38 @@ public Set<Sequence> getAllSequences() {
     return this.allSequences;
   }
 
+  /**
+   * Implements the "GRT Elephant-Brain" component, as described in "GRT: Program-Analysis-Guided
+   * Random Testing" by Ma et. al (ASE 2015): https://people.kth.se/~artho/papers/lei-ase2015.pdf.
+   * If the dynamic type of the last object (output) in the sequence is a subtype of its static
+   * type, cast it to its dynamic type. This allows Randoop to create input objects that cannot be
+   * created using static type information alone for the method under test.
+   *
+   * @param eSeq the executable sequence to apply dynamic type casting on
+   */
+  private void applyDynamicTypeCast(ExecutableSequence eSeq) {
+    List<ReferenceValue> lastValues = eSeq.getLastStatementValues();
+    if (lastValues.isEmpty()) {
+      return;
+    }
+    ReferenceValue lastValue = lastValues.get(0);
+    Type expectedType = lastValue.getType();
+    Type actualType = Type.forClass(lastValue.getObjectValue().getClass());
+
+    if (actualType.isSubtypeOf(expectedType) && !actualType.equals(expectedType)) {
+      TypedOperation castOperation = TypedOperation.createCast(expectedType, actualType);
+
+      // Get the last variable in the sequence that has the expected type, cast it to the actual
```

Comment:
Don't we know this is the first variable in the last statement in the sequence?  Is there a more straightforward way to get it than making a list of all the variables of the given type (which seems like unnecessary work)?